src/game/logic/object/update/stealthupdate.h

```diff
@@ -0,0 +1,89 @@
+/**
+ * @file
+ *
+ * @author Jonathan Wilson
+ *
+ * @brief Stealth Update
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#pragma once
+#include "always.h"
+#include "updatemodule.h"
+class FXList;
+
+class StealthUpdateModuleData : public UpdateModuleData
+{
+public:
+    virtual ~StealthUpdateModuleData() override;
+
+private:
+    BitFlags<OBJECT_STATUS_COUNT> m_hintDetectableConditions;
+    BitFlags<OBJECT_STATUS_COUNT> m_requiredStatus;
+    BitFlags<OBJECT_STATUS_COUNT> m_forbiddenStatus;
+    FXList *m_disguiseRevealFX;
+    FXList *m_disguiseFX;
+    float m_moveThresholdSpeed;
+    float m_friendlyOpacityMin;
+    float m_friendlyOpacityMax;
+    float m_revealDistanceFromTarget;
+    unsigned int m_disguiseTransitionTime;
+    unsigned int m_disguiseRevealTransitionTime;
+    unsigned int m_pulseFrequency;
+    unsigned int m_stealthDelay;
+    int m_stealthLevel;
+    unsigned int m_blackMarketCheckDelay;
+    int m_enemyDetectionEvaEvent;
+    int m_ownDetectionEvaEvent;
+    bool m_innateStealth;
+    bool m_orderIdleEnemiesToAttackMeUponReveal;
+    bool m_disguisesAsTeam;
+    bool m_useRiderStealth;
+    bool m_grantedBySpecialPower;
+    friend class StealthUpdate;

```

Comment:
Can we move the friend to the very top? Down here by the members it is kind of obscure.

Reply:
fixed.